packages/grafana-flamegraph/src/FlameGraphContainer.tsx

```diff
@@ -206,24 +206,13 @@ const FlameGraphContainer = ({
 };
 
 function useColorScheme(dataContainer: FlameGraphDataContainer | undefined) {
-  const [colorScheme, setColorScheme] = useState<ColorScheme | ColorSchemeDiff>(
-    dataContainer?.isDiffFlamegraph() ? ColorSchemeDiff.Default : ColorScheme.ValueBased
-  );
-  useEffect(() => {
-    if (
-      dataContainer?.isDiffFlamegraph() &&
-      (colorScheme === ColorScheme.ValueBased || colorScheme === ColorScheme.PackageBased)
-    ) {
-      setColorScheme(ColorSchemeDiff.Default);
-    }
+  const defaultColorScheme = dataContainer?.isDiffFlamegraph() ? ColorSchemeDiff.Default : ColorScheme.PackageBased;
+  const [colorScheme, setColorScheme] = useState<ColorScheme | ColorSchemeDiff>(defaultColorScheme);
 
-    if (
-      !dataContainer?.isDiffFlamegraph() &&
-      (colorScheme === ColorSchemeDiff.Default || colorScheme === ColorSchemeDiff.DiffColorBlind)
-    ) {
-      setColorScheme(ColorScheme.ValueBased);
-    }
-  }, [dataContainer, colorScheme]);
+  // This makes sure that if we change the data to/from diff profile we reset the color scheme.
+  useEffect(() => {
```

Comment:
Realized that this part of code that deals with changing the mode if we go between diff/nondiff can be simplified a lot.